File: output/ChatGPT/avatar/Python/C++/atcoder_ABC167_C.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>
using namespace std;

int main() {
    int n, m, x;
    cin >> n >> m >> x;

    vector<vector<int>> c(n, vector<int>(m));
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            cin >> c[i][j];
        }
    }

    vector<int> pre(n);
    iota(pre.begin(), pre.end(), 0);

    vector<vector<int>> l;
    vector<int> rem;
    int ans = 0;

    for (int i = 1; i <= n; i++) {
        vector<int> indices(i);
        iota(indices.begin(), indices.end(), 0);
        do {
            l.push_back(indices);
        } while (next_permutation(indices.begin(), indices.end()));
    }

    for (int i = 1; i <= m; i++) {
        for (int j = 0; j < l.size(); j++) {
            int ca = 0;
            for (int ii : l[j]) {
                ca += c[ii][i];
            }
            if (ca < x) {
                rem.insert(rem.begin(), j);
            }
        }

        if (rem.size() > 0) {
            for (int j : rem) {
                l.erase(l.begin() + j);
            }
            rem.clear();
        }

        if (l.empty()) {
            cout << -1 << endl;
            break;
        }
    }
    else {
        for (vector<int>& i : l) {
            int ca = 0;
            for (int j : i) {
                ca += c[j][0];
            }
            if (ans == 0) {
                ans = ca;
            }
            else {
                ans = min(ans, ca);
            }
        }
        cout << ans << endl;
    }


    return 0;
}

File: output/CodeGen/avatar/Python/Java_source/atcoder_ABC167_C.java
public class atcoder_ABC167_C {
    public int minSubArrayLen(int s, int[] nums) {
        int[] c = new int[nums.length];
        int[] pre = new int[nums.length];
        int[] l = new int[nums.length];
        int[] rem = new int[nums.length];
        int ans = 0;
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            c[i] = nums[i];
            pre[i] = i;
        }
        for (int i = 1; i < s + 1; i++) {
            int ca = 0;
            for (int j = 0; j < n; j++) {
                ca += c[j];
            }
            if (ca < s) {
                rem[0] = j;
            } else {
                rem[0] = -1;
            }
            for (int j = 1; j < n; j++) {
                ca = 0;
                for (int k = 0; k < rem[j - 1]; k++) {
                    ca += c[k];
                }
                if (ca < s) {
                    rem[j] = j;
                } else {
                    rem[j] = -1;
                }
            }
            if (rem[n - 1]!= -1) {
                for (int j = rem[n - 1]; j < n; j++) {
                    l[j] = j;
                }
                for (int j = 0; j < n; j++) {
                    if (rem[j]!= -1) {
                        l[rem[j]] = rem[j];
                    }
                }
                for (int j = 0; j < n; j++) {
                    if (l[j]!= j) {
                        l[j] = l[l[j]];
                    }
                }
                for (int j = 0; j < n; j++) {
                    if (l[j]!= j) {
                        rem[j] = l[j];
                    }
                }
            }
        }
        for (int i = 0; i < n; i++) {
            ca = 0;
            for (int j = 0; j < n; j++) {
                ca += c[j];
            }
            if (ans == 0) {
                ans = ca;
            } else {
                ans = Math.min(ans, ca);
            }
        }
        return ans;
    }
}